cocos/core/gfx/webgl2/webgl2-commands.ts

```diff
@@ -1268,35 +1277,51 @@ export function WebGL2CmdFuncCreateSampler (device: WebGL2Device, gpuSampler: IW
         gpuSampler.glWrapT = WebGLWraps[gpuSampler.addressV];
         gpuSampler.glWrapR = WebGLWraps[gpuSampler.addressW];
 
-        gpuSampler.glSampler = glSampler;
+        gpuSampler.glSamplers.set(samplerHash, glSampler);
+
         gl.samplerParameteri(glSampler, gl.TEXTURE_MIN_FILTER, gpuSampler.glMinFilter);
         gl.samplerParameteri(glSampler, gl.TEXTURE_MAG_FILTER, gpuSampler.glMagFilter);
         gl.samplerParameteri(glSampler, gl.TEXTURE_WRAP_S, gpuSampler.glWrapS);
         gl.samplerParameteri(glSampler, gl.TEXTURE_WRAP_T, gpuSampler.glWrapT);
         gl.samplerParameteri(glSampler, gl.TEXTURE_WRAP_R, gpuSampler.glWrapR);
-        gl.samplerParameterf(glSampler, gl.TEXTURE_MIN_LOD, 0);
-        gl.samplerParameterf(glSampler, gl.TEXTURE_MAX_LOD, 1000);
+        gl.samplerParameterf(glSampler, gl.TEXTURE_MIN_LOD, minLod);
+        gl.samplerParameterf(glSampler, gl.TEXTURE_MAX_LOD, maxLod);
     }
 }
 
 export function WebGL2CmdFuncDestroySampler (device: WebGL2Device, gpuSampler: IWebGL2GPUSampler) {
     const { gl } = device;
-    if (gpuSampler.glSampler) {
-        gl.deleteSampler(gpuSampler.glSampler);
+    const it = gpuSampler.glSamplers.values();
+    const res = it.next();
+
+    while (!res.done) {
+        gl.deleteSampler(res.value);
+
         const glSamplerUnits = device.stateCache.glSamplerUnits;
         for (let i = 0; i < glSamplerUnits.length; ++i) {
-            if (glSamplerUnits[i] === gpuSampler.glSampler) {
+            if (glSamplerUnits[i] === res.value) {
                 gl.bindSampler(i, null);
                 glSamplerUnits[i] = null;
             }
         }
-        gpuSampler.glSampler = null;
     }
+
+    gpuSampler.glSamplers.clear();
+}
+
+function getGLSampler (device: WebGL2Device, gpuSampler: IWebGL2GPUSampler, minLod: number, maxLod: number) : WebGLSampler {
+    // better if WebGL2CmdFuncCreateSampler can return value
```

Comment:
just remove the `WebGL2CmdFuncCreateSampler` function and implement everything in `IWebGL2GPUSampler.getGLSampler` instead